aws/aws.go

```diff
@@ -363,19 +372,44 @@ func deleteCFStacks(cfClients map[string]cfClient, stacks []*types.Stack) []erro
 	errChan := make(chan error)
 
 	for r, s := range regionCFStacks {
-		go func(cfClient cfClient, region string, stacks []*types.Stack) {
+		go func(cfClient cfClient, rdsClient rdsClient, region string, stacks []*types.Stack) {
 			defer wg.Done()
 			for _, stack := range stacks {
 				if ctx.DryRun {
 					log.Infof("[AWS] Dry-run set, CloudFormation stack is not deleted: %s, region: %s", stack.Name, region)
 				} else {
+					dbStackResource, err := cfClient.DescribeStackResource(&cloudformation.DescribeStackResourceInput{
+						StackName:         &stack.Name,
+						LogicalResourceId: aws.String("DBInstance"),
+					})
+					if err != nil {
```

Comment:
This two ifs can be merge.

Reply:
Welcome back

Reply:
We need the 2 ifs as there are 3 cases:

1. There was an error "Resource DBInstance does not exist for stack" - we should skip the disable DeletionProtection, but we should delete the stack
2. There was an other type of error - we should skip the delete alltogether and continue to the next stack
3. There were no error meaning we have the DBInstance - we should disable DeletionProtection and then delete the stack

Reply:
Sorry my bad, i misunderstud the position of last closing bracket. If than If is always a sign!

Anyway This does the same with fewer if-s
```
if err == nil {
    dbInstanceId := dbStackResource.StackResourceDetail.PhysicalResourceId
    log.Infof("[AWS] Disabling DeletionProtection for DB instance: %s, region: %s", *dbInstanceId, region)

    _, err := rdsClient.ModifyDBInstance(&rds.ModifyDBInstanceInput{
        DBInstanceIdentifier: dbInstanceId,
        DeletionProtection:   aws.Bool(false),
    })
    if err != nil {
        log.Errorf("[AWS] Skipping stack %s delete because failed to disable DeletionProtection on db instance: %s, err: %s", stack.Name, *dbInstanceId, err)
        errChan <- err
        continue
    }
} else if !strings.Contains(err.Error(), "Resource DBInstance does not exist for stack") {
    log.Errorf("[AWS] Failed to fetch the CloudFormation stack resources for stack: %s, err: %s", stack.Name, err)
    errChan <- err
    continue
}
```

Reply:
Your solution is more readable so I modified the code, thanks!